src/components/Dropdown/index.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+import classNames from 'classnames';
+import dropdownStyles from './styles';
+
+interface DropdownOption {
+  id: number,
+  value: string
+}
+
+interface DropdownProps {
+  name: string,
+  options: DropdownOption[],
+  title: string,
+  className?: string,
+}
+
+const renderOptions = (options: DropdownOption[]) => options.map((opt) => (
+  <MenuItem key={opt.id} value={opt.id}>{opt.value}</MenuItem>
+));
+
+const Dropdown: React.FC<DropdownProps> = (props) => {
+
+  const { className, name, options, title } = props;
+  const [dropdownValue, setDropdownValue] = React.useState<string | number>();
+  const classes = dropdownStyles();
+  const rootClass: string = classNames(classes.root, className);
+
+  const handleChange = (event: React.ChangeEvent<{ value: any }>) => { 
+    setDropdownValue(event.target.value);
+  };
+
+  return (
+    <div className={rootClass}>
+      <FormControl className={classes.formControl}>
+        <InputLabel id={`select-${name}-label`}>{title}</InputLabel>
+        <Select
+          id={`select-${name}`}
+          labelId={`select-${name}-label`}
+          onChange={handleChange}
+          value={dropdownValue}
+        >
+          {options && options.length && renderOptions(options)}
```

Comment:
Other thing I don't like is include logic in the jsx render. We can avoid this by creating a const or a variable who makes this logic already and we compose it here. 

(React uses NULL as a fallback to NOT TO render the component, so that should be  the fallback rather than undefined)